### PR TITLE
Utility to collect per-script info on punctuation from cldr.

### PR DESCRIPTION
**Pull request by dougfelt**
This originally spit out code, it now collects a dict that is used
directly by clients.  Left the fn that generates code in place.

### CODE REVIEW

**Comment by marekjez86** [comment]
LGTM

On Thu, Mar 3, 2016 at 1:23 PM, dougfelt <notifications@github.com> wrote:

> This originally spit out code, it now collects a dict that is used
> directly by clients. Left the fn that generates code in place.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/googlei18n/nototools/pull/186
> Commit Summary
>
>    - Utility to collect per-script info on punctuation from cldr.
>
> File Changes
>
>    - *A* nototools/collect_cldr_punct.py
>    <https://github.com/googlei18n/nototools/pull/186/files#diff-0> (153)
>
> Patch Links:
>
>    - https://github.com/googlei18n/nototools/pull/186.patch
>    - https://github.com/googlei18n/nototools/pull/186.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/nototools/pull/186>.
>



-- 

Marek Z Jeziorek [ 老马 ] | marekj@google.com | 312 725-6958

File: nototools/collect_cldr_punct.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+from nototools import tool_utils
+from nototools import unicode_data
+
+TOOLS_DIR = path.abspath(path.join(path.dirname(__file__), os.pardir))
+CLDR_DIR = path.join(TOOLS_DIR, 'third_party', 'cldr')
+
+def _add_text(chars, text):
+  skip = False
+  for i in range(len(text)):
+    cp = text[i]
```
Comment (jamesgk) [inline nototools/collect_cldr_punct.py]: Could be `for i, cp in enumerate(text):`

File: nototools/collect_cldr_punct.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    _script_to_punct = _build_script_to_punct()
+  return _script_to_punct
+
+
+def _write_script_to_punct(script_to_punct):
+  print 'SCRIPT_TO_PUNCT = {'
+  for script in sorted(script_to_punct):
+    chars = script_to_punct[script]
+    int_chars = [ord(cp) for cp in chars]
+    print "  # %s" % ('|'.join(sorted(chars)))
```
Comment (jamesgk) [inline nototools/collect_cldr_punct.py]: Probably prefer single quotes here for consistency.

**Comment by jamesgk** [comment]
LGTM

File: nototools/collect_cldr_punct.py (hunk 1 above)
Comment (dougfelt) [inline nototools/collect_cldr_punct.py]: Ok.  I keep forgetting about enumerate().

File: nototools/collect_cldr_punct.py (hunk 2 above)
Comment (dougfelt) [inline nototools/collect_cldr_punct.py]: Yep.

**Pull request merged**